interfaces/mount/spec_test.go

```diff
@@ -164,6 +164,23 @@ func (s *specSuite) TestMountEntryFromLayout(c *C) {
 	})
 }
 
+func (s *specSuite) TestParallelInstanceMountEntryFromLayout(c *C) {
+	snapInfo := snaptest.MockInfo(c, snapWithLayout, &snap.SideInfo{Revision: snap.R(42)})
+	snapInfo.InstanceKey = "instance"
+	s.spec.AddLayout(snapInfo)
+	s.spec.AddOvername(snapInfo)
+	c.Assert(s.spec.MountEntries(), DeepEquals, []osutil.MountEntry{
+		// Parallel instance mappings come first
+		{Dir: "/snap/vanguard", Name: "/snap/vanguard_instance", Options: []string{"rbind", "x-snapd.origin=overname"}},
+		{Dir: "/var/snap/vanguard", Name: "/var/snap/vanguard_instance", Options: []string{"rbind", "x-snapd.origin=overname"}},
+		// Layout result is sorted by mount path.
+		{Dir: "/etc/foo.conf", Name: "/snap/vanguard/42/foo.conf", Options: []string{"bind", "rw", "x-snapd.kind=file", "x-snapd.origin=layout"}},
+		{Dir: "/mylink", Options: []string{"x-snapd.kind=symlink", "x-snapd.symlink=/snap/vanguard/42/link/target", "x-snapd.origin=layout"}},
+		{Dir: "/mytmp", Name: "tmpfs", Type: "tmpfs", Options: []string{"x-snapd.mode=01777", "x-snapd.origin=layout"}},
+		{Dir: "/usr", Name: "/snap/vanguard/42/usr", Options: []string{"rbind", "rw", "x-snapd.origin=layout"}},
+	})
```

Comment:
If you agree with my suggestion above please add a test for the per-user mount profile here.